GVRf/Framework/src/org/gearvrf/GVRRenderData.java

```diff
@@ -323,6 +323,31 @@ public boolean getAlphaBlend() {
     public void setAlphaBlend(boolean alphaBlend) {
         NativeRenderData.setAlphaBlend(getPtr(), alphaBlend);
     }
+
+    /**
+     * @return The OpenGL draw mode (e.g. GL_TRIANGLES).
+     */
+    public int getDrawMode() {
+        return NativeRenderData.getDrawMode(getPtr());
+    }
+
+    /**
+     * Set the draw mode for this mesh. Default is GL_TRIANGLES.
+     * 
+     * @param drawMode
+     */
+    public void setDrawMode(int drawMode) {
+        if (drawMode != GL_POINTS || drawMode != GL_LINES
+                || drawMode != GL_LINE_STRIP || drawMode != GL_LINE_LOOP
+                || drawMode != GL_TRIANGLES || drawMode != GL_TRIANGLE_STRIP
+                || drawMode != GL_TRIANGLE_FAN) {
+            throw new IllegalArgumentException(
```

Comment:
OK. I tried so many times and finally found the logic problem here. I think you should use "&&" not "||" here...